### PR TITLE
Fixed #1272 Missing data of attachment while pushing attachments

### LINKED ISSUE #1272: 400 Missing data of attachment while pushing attachments
Hi, 
When I add multiple attachments to the same document, the first revision synced to the sync gateway and when I started to sync the rest of the updates that has new attachments the sync gateway stopped receiving this document with this error `_bulk_docs got an error: 400 Missing data of attachment `.  
The issue exists on the master branch and appeared in version 1.2.0, I have tried version 1.1.0 and the issue didn't appear in old versions of couchbase lite.

Thanks

### PR DESCRIPTION
**Pull request by ashihaby**

### CODE REVIEW

**Comment by ashihaby** [comment]
#1273

**Comment by hideki** [comment]
Hi @ashihaby,

Thank you very much for your PR.

Could you please write unit-test under https://github.com/couchbase/couchbase-lite-android/tree/master/src/androidTest/java/com/couchbase/lite/syncgateway ?

**Comment by ashihaby** [comment]
@hideki Ok I will write test case for that

**Comment by ashihaby** [comment]
@hideki please check test case here couchbase/couchbase-lite-android#931

**Comment by hideki** [comment]
@ashihaby, Thank you for writing test case. I will review it soon.

File: src/main/java/com/couchbase/lite/Database.java [hunk 1]
```diff
@@ -906,10 +906,8 @@ public boolean expandAttachments(final RevisionInternal rev,
 
                 int revPos = (Integer) attachment.get("revpos");
                 if (revPos < minRevPos && revPos != 0) {
-                    Map<String, Object> map = new HashMap<String, Object>();
-                    map.put("stub", true);
-                    map.put("revpos", revPos);
-                    return map;
+                    //nothing changed
+                    return attachment;
```
Comment (hideki) [inline src/main/java/com/couchbase/lite/Database.java]: Hi @ashihaby,
Is there any test scenario that `stub` does not work?

**Comment by hideki** [comment]
Hi @ashihaby,

I think changes for Database.java is unnecessary. So I will merge only changes for PushInternal.java. If you know test case that is caused by `stub` attachment.  Please file another ticket?

**Comment by hideki** [comment]
Hi @ashihaby,
Thank you very much for another contribution. And  I am sorry it took long time to review your codes. I believe Couchbase Lite becomes more stable with your help.
Thanks again!

**Pull request merged**